test_scripts/pypeit_test.py

```diff
@@ -0,0 +1,1159 @@
+    else:
+        instruments = np.array([item for item in all_instruments 
+                                    if pargs.tests.lower() in item.lower()])
+        unsupported = [item for item in instruments 
+                            if not np.any([inst.lower() in item.lower()
+                                for inst in supported_instruments()]) ]
+
+    # Check that instruments is not blank
+    if len(instruments) == 0:
+        print("\x1B[" + "1;31m" + "\nERROR - " + "\x1B[" + "0m" +
+              "Invalid test selected: {0:s}\n\n".format(pargs.tests) +
+              "Consult the help (pypeit_test -h) or select one of the " +
+              "available RAW_DATA directories: {0}".format(', '.join(all_instruments)))
+        return 1
+
+    if len(unsupported) > 0:
+        if not pargs.quiet:
+            print("\x1B[" + "1;33m" + "\nWARNING - " + "\x1B[" + "0m" +
+                  "The following tests have not been validated and may not pass: {0}\n\n".format(
+                  unsupported))
+
+    if not os.path.exists(pargs.outputdir):
+        os.mkdir(pargs.outputdir)
+    outputdir = os.path.abspath(pargs.outputdir)
+
+    # Make sure we can create a writable report file (if needed) before starting the tests
+    if pargs.report is None and pargs.quiet:
+        # If there's no report file specified in the command line, but we're in quiet mode,
+        # make up a report file name
+        pargs.report = get_unique_file(os.path.join(outputdir, "pypeit_test_results.txt"))
+
+    if pargs.report:
+        try:
+            report_file = open(pargs.report, "w")
+        except Exception as e:
+            print(f"Could not open report file {report_file}", file=stderr)
+            traceback.print_exc()
+            sys.exit(1)
+
+
+    # Report
+    if not pargs.quiet:
+        print('Running tests on the following instruments:')
+        for instr in instruments:
+            print('    {0}'.format(instr))
+        print('')
+
+    #---------------------------------------------------------------------------
+    # Check all the data and relevant files exists before starting!
+    # TODO: Do this for "all", as well?
+    if pargs.tests in tests_that_only_use_dev_setups:
+        unsupported = []
+        missing_data = []
+        missing_pypfiles = []
+        for instr in instruments:
+            # Only do blue instruments
+            if pargs.debug and 'blue' not in instr:
+                continue
+            if instr not in devsetups.keys():
+                # TODO: We should never get here, right?
+                unsupported += [ instr ]
+
+            # Setups
+            setups = next(os.walk(os.path.join(raw_data, instr)))[1]
+
+            # Check all data before starting
+            for setup in devsetups[instr]:
+                if setup not in setups:
+                    missing_data += [ '{0}/{1}'.format(instr, setup) ]
+
+                if instr == 'shane_kast_blue' and '600' in setup:
```

Comment:
Can we generalize this?  By that, I mean add to one of the classes above a set of `setup` instruments.  And we may add to that if we want to do this for others?  As opposed to this hack I put in long ago..